javascript/word-count.js

```diff
@@ -0,0 +1,29 @@
+var Words = function() {
+  this.count = function(data){
+    var resultHash = {};
+
+    // split the input string on one or more instances of
+    // whitespace, tab, or newline
+    wordsArray = data.trim().split(/[\n\t\s]+/);
+
+    for (i=0; i < wordsArray.length; i++){
+
+      // Check for string exists as property on object and, if so, send that
+      // value as an array to allow it to be treated as a string literal.
+      if(typeof(resultHash[wordsArray[i]]) == "function"){
```

Comment:
This `typeof(resultHash[wordsArray[i]]) == "function"` matcher is confusing. Is there a more straightforward way to write the entire if else block to get rid of the nested conditionals?